src/Microsoft.SqlTools.ManagedBatchParser/ReliableConnection/RetryPolicyUtils.cs

```diff
@@ -357,6 +377,38 @@ public static void RaiseSchemaAmbientRetryMessage(RetryState retryState, int err
             }
         }
 
+        /// <summary>
+        /// Wait for SqlConnection to handle sleeping serverless instances (allows for them to wake up).
+        /// </summary>
+        public static SqlRetryLogicBaseProvider ServerlessWaitRetryLogicProvider()
+        {
+            var serverlessRetryLogic = new SqlRetryLogicOption
+            {
+                NumberOfTries = _serverlessMaxRetries,
+                MaxTimeInterval = TimeSpan.FromSeconds(_serverlessMaxIntervalTime),
+                DeltaTime = TimeSpan.FromSeconds(1),
+                TransientErrors = _retryableExtendedNetworkConnectivityError
+            };
+
+            var provider = SqlConfigurableRetryFactory.CreateFixedRetryProvider(serverlessRetryLogic);
+
+            provider.Retrying += (object s, SqlRetryingEventArgs e) =>
+            {
+                int attempts = e.RetryCount + 1;
```

Comment:
I do not know what the convention for this repo (or file) is, but... since you used `var` above (which I like), keep using it :-)